### ISSUE TITLE
portal device inverter 

### ISSUE THREAD

**Issue by BETTEREXTENEDEDITORMOD** [BEEmod/BEE2-items#2246]
this is the PDI testing element from the workshop map [](url)https://steamcommunity.com/sharedfiles/filedetails/?id=327359835 this inverts what portals you shoot when you walk through it. you can use the fizzler and laser field options to chose what portal color it starts at

**Comment by ZombiePwner1** [comment]
I'm not too sure if this would be possible in the puzzlemaker but ill let @TeamSpen210 decide.

**Comment by TeamSpen210** [comment]
What does the device do exactly?

**Comment by BETTEREXTENEDEDITORMOD** [comment]
it changes what portal the gun can shoot if present in a map your gun could not shoot a portal until you walk through it 
if you walk through it will for example give you a blue portal if you walk through it again an orange portal it does not matter what side you'l enter it will always give you the opposite portal to what your wielding the fizzler/laser field options will change what portal you'l initially get 
this fizzler will change color to show what portal you'l get if it shows orange for example you will get a orange portal but if you walk through it it'l change blue to show you what portal you'l get when you walk through it again 
<img width="446" alt="capture 44" src="https://user-images.githubusercontent.com/31172185/31107478-8ce9395e-a7a7-11e7-845e-1a69325ad363.PNG">
all your portals do not fizzle when you walk through a fizzler only the portal that you can shoot

**Comment by IceTudor** [comment]
Interesting, I agree, it should be added (says me, the one making a level series based on the custom fizzlers)

**Comment by BETTEREXTENEDEDITORMOD** [comment]
it would be very simple to implement from what i can tell it just uses triggers to play the sound when you walk through and to change what portal you can shoot and also the color 
<img width="243" alt="capture 48" src="https://user-images.githubusercontent.com/31172185/31322192-1db0f112-ac48-11e7-9cd5-d454d8860f3a.PNG">

**Comment by TeamSpen210** [comment]
No, it's not. How would it function with autoportals in the map?

**Comment by BETTEREXTENEDEDITORMOD** [comment]
i imagine for example if you get the blue portal the orange auto portal will open and vice versa

**Comment by Endermage77** [comment]
I'd think it'd just leave the portals there until the portal gun overwrites them.

**Comment by TeamSpen210** [comment]
That's not really possible, it's 'dangerous' if the player is able to place a portal of the same colour as an autoportal, they end up taking control of the autoportal one, making all manner of weirdness.
